File: service-discovery-client/src/main/java/com/alekseysamoylov/servicediscoveryclient/config/ClientConfiguration.java
package com.alekseysamoylov.servicediscoveryclient.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.cloud.context.config.annotation.RefreshScope;
import org.springframework.stereotype.Component;

@Component
@RefreshScope
public class ClientConfiguration {

    @Value("${configurable-name}")
    private String configurableName;

    public String getConfigurableName() {
        return configurableName;
    }
}

File: kotlin-client/src/main/kotlin/com/alekseysamoylov/kotlinclient/config/ClientConfiguration.kt
package com.alekseysamoylov.kotlinclient.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.cloud.context.config.annotation.RefreshScope
import org.springframework.stereotype.Component

@Component
@RefreshScope
class ClientConfiguration {

    @Value("\${configurable-name}")
    private lateinit var configurableName: String

    fun getConfigurableName(): String {
        return configurableName
    }
}
